### PR TITLE
Check if users do not get custom editor

### PR DESCRIPTION
**Pull request by DonJayamanne**
For https://github.com/microsoft/vscode-jupyter/issues/5281

### CODE REVIEW

File: src/client/common/application/customEditorService.ts [hunk 1]
```diff
@@ -44,35 +44,43 @@ export class CustomEditorService implements ICustomEditorService {
...
+                Array.isArray(editorAssociations) &&
+                editorAssociations.find((item) => item.viewType === ViewType)
+            ) {
+                const updatedSettings = editorAssociations.filter((item) => item.viewType !== ViewType);
+                updateType = 'removed';
+                await settings.update('editorAssociations', updatedSettings, vscode.ConfigurationTarget.Global);
+                sendTelemetryEvent(Telemetry.UpdateCustomEditorAssociation, undefined, { type: 'removed' });
+            }
+        } catch (ex) {
+            sendTelemetryEvent(Telemetry.UpdateCustomEditorAssociation, undefined, { type: updateType! }, ex, true);
```
Comment (DonJayamanne) [inline src/client/common/application/customEditorService.ts:83]: At least 2 users ran into this issue and could not open notebooks. Possible there are more users.

File: src/client/common/application/customEditorService.ts [hunk 2]
```diff
@@ -44,35 +44,43 @@ export class CustomEditorService implements ICustomEditorService {
...
-        // This code is temporary.
-        const settings = this.workspace.getConfiguration('workbench', undefined);
-        const editorAssociations = settings.get('editorAssociations') as {
-            viewType: string;
-            filenamePattern: string;
-        }[];
+    public async activate() {
+        let updateType: 'added' | 'removed' | undefined;
+        try {
+            // This code is temporary.
```
Comment (IanMatthewHuff) [inline src/client/common/application/customEditorService.ts:50]: Apparently not!

**Review by IanMatthewHuff — APPROVED** [review]

File: src/client/common/application/customEditorService.ts (hunk 2 above)
Comment (DonJayamanne) [reply src/client/common/application/customEditorService.ts:50]: Lol, i guess temporary for the long term  😆

**Comment by heavywatal** [comment]
Is it possible to achieve this behavior without violating the user's `settings.json`? Opening `*.ipynb` in `jupyter.notebook.ipynb` is not my customization, but vscode-jupyter's default behavior.

**Pull request merged**